Mathlib/LinearAlgebra/TensorProduct/Finiteness.lean

```diff
@@ -0,0 +1,301 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.LinearAlgebra.TensorProduct.Basic
+import Mathlib.RingTheory.Finiteness
+
+/-!
+
+# Some finiteness results of tensor product
+
+This file contains some finiteness results of tensor product.
+
+- `TensorProduct.exists_multiset`, `TensorProduct.exists_finsupp_left`,
+  `TensorProduct.exists_finsupp_right`, `TensorProduct.exists_finset`:
+  any element of `M ⊗[R] N` can be written as a finite sum of pure tensors.
+  See also `TensorProduct.span_tmul_eq_top`.
+
+- `TensorProduct.exists_finite_submodule_left_of_fintype`,
+  `TensorProduct.exists_finite_submodule_right_of_fintype`,
+  `TensorProduct.exists_finite_submodule_of_fintype` and 21 more:
+  any finitely many elements `M ⊗[R] N` is contained in `M' ⊗[R] N'`
+  for some finitely generated submodules `M'` and `N'` of `M` and `N`, respectively.
+  Each of these 3 functions has 8 variants.
+
+## Tags
+
+tensor product, finitely generated
+
+-/
+
+open scoped Classical TensorProduct
```

Comment:
```suggestion
open scoped TensorProduct
```
This breaks a couple of proofs, but it is enough to start them with `classical`.